src/OpenColorIO/ops/Lut1D/Lut1DOpData.h

```diff
@@ -0,0 +1,334 @@
+/*
+Copyright (c) 2018 Autodesk Inc., et al.
+All Rights Reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are
+met:
+* Redistributions of source code must retain the above copyright
+  notice, this list of conditions and the following disclaimer.
+* Redistributions in binary form must reproduce the above copyright
+  notice, this list of conditions and the following disclaimer in the
+  documentation and/or other materials provided with the distribution.
+* Neither the name of Sony Pictures Imageworks nor the names of its
+  contributors may be used to endorse or promote products derived from
+  this software without specific prior written permission.
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#ifndef INCLUDED_OCIO_LUT1DOPDATA_H
+#define INCLUDED_OCIO_LUT1DOPDATA_H
+
+#include <OpenColorIO/OpenColorIO.h>
+
+#include "Op.h"
+#include "ops/OpArray.h"
+
+OCIO_NAMESPACE_ENTER
+{
+class Lut1DOpData;
+typedef OCIO_SHARED_PTR<Lut1DOpData> Lut1DOpDataRcPtr;
+typedef OCIO_SHARED_PTR<const Lut1DOpData> ConstLut1DOpDataRcPtr;
+
+class Lut1DOpData : public OpData
+{
+public:
+
+    // List of flags that describe 1-D LUT index and value encoding.
+    //
+    // 1-D LUT indices and values can either be expressed in standard numeric
+    // encodings or using half float codes.   Half float codes are 16 bit integer
+    // representations of a 16 bit floating point value. See:
+    // http://en.wikipedia.org/wiki/Half-precision_floating-point_format.
+    //
+    enum HalfFlags {
+        LUT_STANDARD = 0x00,         // Indices & values use standard encoding.
+        LUT_INPUT_HALF_CODE = 0x01,  // LUT indices are half float codes.
+        LUT_OUTPUT_HALF_CODE = 0x02, // LUT values are half float codes.
+
+        LUT_INPUT_OUTPUT_HALF_CODE =
+        LUT_INPUT_HALF_CODE
+        | LUT_OUTPUT_HALF_CODE       // Indices and values are half float codes.
+    };
+
+    // Enum to control optional hue restoration algorithm.
+    enum HueAdjust
+    {
+        HUE_NONE = 0, // No adjustment.
+        HUE_DW3       // Algorithm used in ACES Output Transforms through v0.7.
+    };
+
+    // Enumeration of the inverse 1D LUT styles.
+    enum InvStyle
```

Comment:
I am afraid I don't understand your comment. Is it an issue that we are using the same names for LUT 1D and LUT 3D?

Reply:
Sorry to be vague. I was referring to our discussion in issue #621 

We discussed that InvStyle was not very clear naming, and that InversionQuality was better. However, in thinking through this more, we were discussing the high-level interface there, and this is the lower level OpData interface.

In looking at other OpData classes, I can see that "style" is a convention also being used by Range and CDL. On that point, I'm OK with standing down on this one.

I'm still reviewing this PR; it's non-trivial to navigate because of the WIP nature and some files changing so much that the diffs provide little value. Will hopefully finish today.

Reply:
I think your point is a good one Michael.  The inversion method here is somewhat different than the notion of style in Range and CDL where it is essentially which equation is to use rather than specifying a speed/quality trade-off.  So I think we should use inversionQuality as discussed in #621.  But we still need to do the PR to change the Lut3D so the Lut1D could be adjusted at the same time.  Right now we're just trying to get the needed pieces in to unblock the CLF PR and then will come back to some of these tweaks like the style vs. quality issue.

Thanks for your careful review, much appreciated!